### PR TITLE
Ignore type variable casts in `ClassCastLambdaUsage` check

### PR DESCRIPTION
**Pull request by mohamedsamehsalah**
Thanks for pointing out @rickie 🙏 

Suggested commit message
```
Ignore type variable casts in `ClassCastLambdaUsage` check (#1449)
```

### CODE REVIEW

**Review by rickie — APPROVED** [review]
Actually, it was @EnricSala who pointed this out 😉.

Added a suggestion :).

File: error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/ClassCastLambdaUsage.java [hunk 1]
```diff
@@ -48,7 +49,10 @@ public Description matchLambdaExpression(LambdaExpressionTree tree, VisitorState
     }
 
     Type type = ASTHelpers.getType(typeCast);
-    if (type == null || type.isParameterized() || type.isPrimitive()) {
+    if (type == null
```
Comment (rickie) [inline error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/ClassCastLambdaUsage.java:52]: I'm confused now, was the pitest warning already here or is it new 🤔?

Comment (mohamedsamehsalah) [reply error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/ClassCastLambdaUsage.java:52]: AFAIK:

1. It was in the original commit, pre suggestions.
2. It was introduced with this [PR](https://github.com/PicnicSupermarket/error-prone-support/pull/1385/files), so it is probably referencing mutating the `#isPrimitive` call 🤔

Comment (rickie) [reply error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/ClassCastLambdaUsage.java:52]: Thanks for checking @mohamedsamehsalah 🚀🚀🚀!

Comment (mohamedsamehsalah) [reply error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/ClassCastLambdaUsage.java:52]: Any clue why is this reported? 😅

Comment (mohamedsamehsalah) [reply error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/ClassCastLambdaUsage.java:52]: > Any clue why is this reported? 😅

I re-read the warning line
> A change can be made to line 52 without causing a test to fail

which seems to come from mutating the nullability check; which seems common for similar PRs (for example, [here](https://github.com/PicnicSupermarket/error-prone-support/pull/494/files)).

File: error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/ClassCastLambdaUsage.java [hunk 2]
```diff
@@ -48,7 +49,10 @@ public Description matchLambdaExpression(LambdaExpressionTree tree, VisitorState
...
+        || type.isParameterized()
+        || type.isPrimitive()
+        || type.getKind() == TypeKind.TYPEVAR) {
       /*
        * The method reference syntax does not support casting to parameterized types. Additionally,
        * `Class#cast` does not support the same range of type conversions between (boxed) primitive
```
Comment (Stephan202) [inline error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/ClassCastLambdaUsage.java]: We should update this comment.

File: error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/ClassCastLambdaUsageTest.java [hunk 3]
```diff
@@ -32,13 +32,17 @@ void identification() {
...
+            "    Stream.of(8).reduce((a, b) -> (Integer) a);",
+            "    IntStream.of(9).mapToObj(i -> (char) i);",
             "",
             "    // BUG: Diagnostic contains:",
-            "    Stream.of(8).map(i -> (Integer) i);",
+            "    Stream.of(10).map(i -> (Integer) i);",
+            "  }",
+            "",
+            "  private <T> void m2() {",
+            "    Stream.of(11).map(i -> (T) i);",
```
Comment (Stephan202) [inline error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/ClassCastLambdaUsageTest.java]: We can add the type variable to `m()` instead.

**Review by Stephan202 — APPROVED** [review]
Rebased and added a commit. Tnx @mohamedsamehsalah and @EnricSala!

Suggested commit message:
```
Ignore type variable casts in `ClassCastLambdaUsage` check (#1449)
```

@rickie I'll let you merge if you agree with the changes.

**Comment by rickie** [comment]
Rebased, changes LGTM!

**Pull request merged**